plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMergeSink.java

```diff
@@ -279,8 +280,13 @@ private DataFileInfo rewriteParquetFile(Path path, ImmutableLongBitmapDataProvid
                     page = page.getPositions(retained, 0, retainedCount);
                 }
 
+                rowsWritten += page.getPositionCount();
                 fileWriter.appendRows(page);
```

Comment:
Let's optimize this to do
```java
if (page.getPositionCount() > 0) {
    fileWriter.appendRows(page);
}
```
Since the writer seems to do non-trivial things even when the page is empty.

Reply:
Changed.